test/micro/org/openjdk/bench/jdk/incubator/foreign/ParallelSum.java

```diff
@@ -209,15 +209,19 @@ public int unsafe_parallel() {
         @Override
         protected Integer compute() {
             if (length > SPLIT_THRESHOLD) {
-                SumUnsafe s1 = new SumUnsafe(address, start, length / 2);
-                SumUnsafe s2 = new SumUnsafe(address, length / 2, length / 2);
+                int rem = length % 2;
+                int split = length / 2;
+                int lobound = split;
+                int hibound = lobound + rem;
+                SumUnsafe s1 = new SumUnsafe(address, start, lobound);
+                SumUnsafe s2 = new SumUnsafe(address, start + lobound, hibound);
```

Comment:
Judging by this code the third parameter is no longer a length, but an upper bound? Maybe you also want to change the field name in that case.

Reply:
Uhm - it's still the length (in elements) associated with the task. I'll leave the field name as is.